apps/api/src/routers/user.py

```diff
@@ -105,7 +97,8 @@ async def apply_flow(
     raw_app_data_dump = raw_application_data.model_dump()
 
     for field in ["pronouns", "ethnicity", "school", "major"]:
-        if field in raw_app_data_dump and raw_app_data_dump[field] == "other":
+        value = raw_app_data_dump.get(field)
+        if any([value == "other", isinstance(value, list) and "other" in value]):
```

Comment:
Suggestion: the usage of any when this is not a dynamic iterable seems confusing. Consider replacing with a disjunction: `if cond1 or cond2:`.